config/initializers/cypress_rails.rb

```diff
@@ -1,9 +1,12 @@
-return unless Rails.env.test?
+# We want to check not only if we are in the test environment, but also if we are
+# running E2E tests. Otherwise this will run when system tests run.
+return unless Rails.env.test? && ENV["E2E"].present?
```

Comment:
**comment (non-blocking):** This may be overkill, but I want to point out that Rails allow you to add more environments than just the 3 standard ones. To do you do the create a new file in config environments (e.g. `config/environments/e2e.rb`, probably starting out as a copy of `test.rb`) and add a section in `config/database.yml` and `config/webpacker.yml`. There technically would be a few other files, but we use neither `ActionCable` nor the built-in Rails secret mechanism, so we don't need to update them. You'd also need to add `:e2e` to the relevant groups in the `Gemfile`. Then the custom env can be used like any other, e.g. `Rails.env.e2e?` or `rails console -e e2e`.

Reply:
I think all of this makes sense. As it's non-blocking for the moment, I think I will do this in a subsequent PR.

Reply:
Maintaining an extra environment is a bit of work, so we should only do it if we're uncomfortable with the amount of conditionals, etc. "polluting" the normal test env.

Reply:
In that case, I think the extra conditional is not too much overhead. If we don't have to maintain another environment, probably for the better.